WIP: GenerateCoverage Experiment

danjujan opened the pull request.

vulder commented on tests/helper_utils.py:
```diff
@@ -8,6 +8,7 @@
 from functools import wraps
```
Why do the imports of this file change but no conent?

danjujan replied:
Probably an oversight while I developed https://github.com/se-sic/VaRA-Tool-Suite/pull/770. Removed them.

vulder commented on tests/plots/test_llvm_coverage_plot.py:
```diff
@@ -0,0 +1,127 @@
+
+class TestCodeRegion(unittest.TestCase):
+
+    def test_feature_config_report_map(self):
```
I'm not sure what this tests, the beginning looks a bit like testing the config handling part of the configuration, which should be tested in separation and in the config module.

danjujan replied:
This only tests if `ConfigCoverageReportMapping` will use the right configs for diffing.

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+"""Code region tree and coverage report."""
+
+from __future__ import annotations
```
From which python version is this? As we have update to min 3.9 we maybe don't need this anymore

danjujan replied:
This is still necessary. Otherwise the following is not possible. See https://peps.python.org/pep-0563/ and https://peps.python.org/pep-0649/
```python
class CodeRegion
    def from(cls, x) -> CodeRegion:
       ...
```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+                node.childs.append(region)
+                node.childs.sort()
+                region.parent = node
+                # Actually this is possible,
+                # e.g. a for loop can be executed
+                # more often than its function.
```
```suggestion
                # Actually this is possible, e.g., a for loop can
                # be executed more often than its function.
```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+                # Actually this is possible,
+                # e.g. a for loop can be executed
+                # more often than its function.
+                #assert region.count <= node.count
```
Still needed?

danjujan replied:
No, that was a comment/assertion for me while developing the code region tree. Removed it.

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+        # when traversing the tree in postorder
+        for node in self.iter_postorder():
+            if node.is_subregion(region):
+                if len(node.childs) > 0:
```
```suggestion
                if node.childs:
```
This should be enough, as empty lists are false

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+                    # node is not a leaf node
+                    # check which childs should become childs of regions
+                    childs_to_move = []
+                    for child in node.childs:
+                        if region.is_subregion(child):
+                            childs_to_move.append(child)
+                    region.childs.extend(childs_to_move)
+                    region.childs.sort()
+                    for child in childs_to_move:
+                        child.parent = region
+                        node.childs.remove(child)
```
```suggestion
                    for child in node.childs:
                        if region.is_subregion(child):
                            childs_to_move.append(child)

                    region.childs.extend(childs_to_move)
                    region.childs.sort()

                    for child in childs_to_move:
                        child.parent = region
                        node.childs.remove(child)

```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+        """Merges region into self by adding all counts of region to the counts
+        of self."""
+        for x, y in zip(self.iter_breadth_first(), region.iter_breadth_first()):
+            assert x == y, "CodeRegions are not identical"
```
As far as I now, a PEP suggests to use raise assertion error over assert, to also trigger in non debug cases

danjujan replied:
Changed.

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+
+    def is_identical(self, other: object) -> bool:
+        """Is the code region equal and has the same coverage?"""
+        if not isinstance(other, CodeRegion):
+            return False
+        if not (self == other and self.count == other.count):
+            return False
+        for code_region_a, code_region_b in zip(
```
```suggestion
        if not isinstance(other, CodeRegion):
            return False

        if not (self == other and self.count == other.count):
            return False

        for code_region_a, code_region_b in zip(
```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+                return False
+        return True
+
+    # Compare regions only depending on their
+    # start lines and columns + their type
+
```
This rnd. comment in the class seems a bit lost, should it be put somewhere else?

danjujan replied:
This comment refers to the following comparison functions `__eq__`, `__lt__`, `__gt__`,`__le__` and `__ge__`. I could make it a docstring for `__eq__` instead or simply remove it.

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+                code_region_a == code_region_b and
+                code_region_a.count == code_region_b.count
+            ):
+                return False
+        return True
```
```suggestion
                return False

        return True
```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+    # start lines and columns + their type
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CodeRegion):
+            return False
+        return (
```
```suggestion
        if not isinstance(other, CodeRegion):
            return False

        return (
```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+            self.start.line == other.start.line and
+            self.start.column < other.start.column
+        ):
+            return True
+        return False
```
```suggestion
            return True

        return False
```
I personally, like a few empty lines here and there to separate logical blocks, otherwise it's harder to spot for the eye

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+class FunctionCodeRegionMapping(tp.Dict[str, CodeRegion]):
+    """Mapping from function names to CodeRegion objects."""
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, FunctionCodeRegionMapping):
+            return False
+        for self_value, other_value in zip(self.values(), other.values()):
+            if not self_value.is_identical(other_value):
+                return False
+        return True
```
```suggestion
    def __eq__(self, other: object) -> bool:
        if not isinstance(other, FunctionCodeRegionMapping):
            return False

        for self_value, other_value in zip(self.values(), other.values()):
            if not self_value.is_identical(other_value):
                return False

        return True
```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+                code_region_b = report.filename_function_mapping[filename_b][
+                    function_b]
+                assert code_region_a == code_region_b
+
+                code_region_a.diff(code_region_b)
+
+    def _import_functions(
+        self, json_file: Path,
+        filename_function_mapping: FilenameFunctionMapping
+    ) -> FilenameFunctionMapping:
+        with json_file.open() as file:
+            j = json.load(file)
+        # Compatibility check
+        try:
```
`j` could be undefined, producing a hard to interpret run-time error

danjujan replied:
A `NotImplementedError` is now raised when json decoding fails.

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+            j_type = j["type"]
+            j_version = j["version"].split(".")
+            assert j_type == "llvm.coverage.json.export" and j_version[0] == "2"
+        except Exception as err:
```
Can you make the catch a bit more precise?

danjujan replied:
Done.

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+        filename_function_mapping: FilenameFunctionMapping
+    ) -> FilenameFunctionMapping:
+        with json_file.open() as file:
+            j = json.load(file)
```
Maybe give `j` a nicer more "clear" name

danjujan replied:
Renamed `j` to `coverage_json`

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+
+        for function in functions:
+            name: str = function["name"]
+            # count: int = function["count"]
+            # branches: list = function["branches"]
```
not needed, or why are these outcommented?

danjujan replied:
Not needed for now. These are parts in the json that could be interesting for someone who wants to extend the parsing in the future.

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+
+        absolute_path = j["absolute_path"]
+        data: tp.Dict[str, tp.Any] = j["data"][0]
+        # files: tp.List = data["files"]
```
not needed?

danjujan replied:
Not needed for now. Same answer as above.

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+                        if region.is_covered():
+                            covered_regions += 1
+                        else:
+                            notcovered_regions += 1
```
```suggestion
                            notcovered_regions += 1

```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+        function_region_mapping = report.filename_function_mapping[file]
+        path = Path(file)
+        tmp_value = _cov_show_file(path, base_dir, function_region_mapping, [])
+        if not tmp_value[-1].endswith("\n"):
+            # Add newline if file does not end with one
+            tmp_value.append("\n")
```
```suggestion

        if not tmp_value[-1].endswith("\n"):
            # Add newline if file does not end with one
            tmp_value.append("\n")

```

vulder commented on varats/varats/data/reports/llvm_coverage_report.py:
```diff
@@ -0,0 +1,708 @@
+                           ) if segments[-1][1].endswith("\n") and (
+                               str.isspace(segments[-1][1].replace(";", "", 1))
+                           ) else segments[-1]
+        #buffer.append(str(segments) + "\n")
```
no longer needed?

danjujan replied:
Correct. Removed.

vulder commented on varats/varats/experiments/vara/llvm_coverage_experiment.py:
```diff
@@ -0,0 +1,174 @@
+                    # Add absolute path to json to compute
+                    # relative filenames later in the report
+                    with open(json_name) as file:
+                        coverage = json.load(file)
+                    coverage["absolute_path"] = str(
+                        Path(self.project.source_of_primary).resolve()
+                    )
```
```suggestion
                        coverage = json.load(file)

                    coverage["absolute_path"] = str(
                        Path(self.project.source_of_primary).resolve()
                    )

```

vulder commented on varats/varats/experiments/vara/llvm_coverage_experiment.py:
```diff
@@ -0,0 +1,174 @@
+        return actions.StepResult.OK
+
+
+# Please take care when changing this file, see docs experiments/just_compile
```
Maybe also add a quite note why?

boehmseb replied:
I think this was copied by mistake. This comment is there to keep the compile experiment in sync with the docs.

danjujan replied:
Yeay, copy pasted from the JustCompile experiment. Removed.

vulder commented on varats/varats/plots/llvm_coverage_plot.py:
```diff
@@ -0,0 +1,323 @@
+        return repr(self.x)
+
+
+class RunConfig(tp.FrozenSet[tp.Tuple[str, ConfigValue]]):
```
Why do you need to build a config class here, is the one that parses the config format not working. If not, can we maybe adapt it?

vulder reviewed (changes requested):
Hmm, I'm not 100% sure about the zip files for the tests. On the one side, it's good for testing, on the other, we don't know how to update/fix them when stuff breaks.
Maybe, we should at least put some doc/wording in a file that specifies how the information can be regenerated?
+ I hope, this does not make the tests brittel

@boehmseb what's your take on that?

Overall, the implementation already looks quite good.

boehmseb commented:
> Hmm, I'm not 100% sure about the zip files for the tests. On the one side, it's good for testing, on the other, we don't know how to update/fix them when stuff breaks. Maybe, we should at least put some doc/wording in a file that specifies how the information can be regenerated?
> 
>     * I hope, this does not make the tests brittle
> 
> 
> @boehmseb what's your take on that?

We kind of brought this on ourselves by introducing .zip files as reports.
The corresponding report class should make obvious why there is a .zip lying around.
So technically, this is no different to including a normal report as test input.
I understand your concern, but I'm not sure whether an additional readme of some sorts can provide additional information that is not already encoded by the test and the report class.

danjujan commented:
> > Hmm, I'm not 100% sure about the zip files for the tests. On the one side, it's good for testing, on the other, we don't know how to update/fix them when stuff breaks. Maybe, we should at least put some doc/wording in a file that specifies how the information can be regenerated?
> > ```
> > * I hope, this does not make the tests brittle
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @boehmseb what's your take on that?
> 
> We kind of brought this on ourselves by introducing .zip files as reports. The corresponding report class should make obvious why there is a .zip lying around. So technically, this is no different to including a normal report as test input. I understand your concern, but I'm not sure whether an additional readme of some sorts can provide additional information that is not already encoded by the test and the report class.

As @boehmseb said. The zip files were obtained by copy pasting them from the results folder after running the `GenerateCoverage` experiment.

vulder commented:
That part I know and "buy", I'm more concerned to how we can update them if needed. We should add docu to the tests how to regenerate the files, project + experiment.
But what also concerns me, it looks like some of the paths where edited by had, which could introduce brittleness as this rewriting is not done automatically. I would be fine with the zips if we have a small description in the test files that tells a developer how to regenerate these files, because from what I see, it's a bit more difficult than "just" running the experiment.

By the way, as soon as we have a "docu format" or idea how to specify these, we should probably also document the other ones.

danjujan commented:
> That part I know and "buy", I'm more concerned to how we can update them if needed. We should add docu to the tests how to regenerate the files, project + experiment. But what also concerns me, it looks like some of the paths where edited by had, which could introduce brittleness as this rewriting is not done automatically. I would be fine with the zips if we have a small description in the test files that tells a developer how to regenerate these files, because from what I see, it's a bit more difficult than "just" running the experiment.
> 
> By the way, as soon as we have a "docu format" or idea how to specify these, we should probably also document the other ones.

I regenerated the files and noted down my steps. https://github.com/se-sic/VaRA-Tool-Suite/blob/fba20d72770514086ea65262a491378c5ea11963/tests/TEST_INPUTS/results/FeaturePerfCSCollection/HowToReproduce.md. Feedback welcome.

danjujan committed:
Diff only available features in plot. Removes unnecessary -(header^slow) or similar lines in plots,

danjujan committed:
Coverage Experiemnt: Compile twice once with VaRAs feature regions annotated, once with profiling instructions.

danjujan committed:
Add test for correctly annotating vara_instrs to code regions

danjujan committed:
Working threshold in confusion matrix

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Fix annotating empty "" feature in norm instr.

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Presence Condition seem to work. Cleanup still required

danjujan committed:
Presence Conditions work. No diffing/merging needed for them

danjujan committed:
Fix mypy + pylint errors

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Fix mypy errors

danjujan committed:
Fix docs generation failing due to missing pyeda import

danjujan committed:
Merge branch 'vara-dev' into jan-dev

vulder committed:
Implements classification evaluation helper

vulder committed:
Updates naming and error handling

vulder committed:
Renaming

vulder committed:
Removes unnecessary import

danjujan committed:
Add pyeda dependency for container

danjujan committed:
Set container image for FeaturePerfCSCollections

danjujan committed:
Start removing merging + diffing code

danjujan committed:
More merging + diffing code removal

danjujan committed:
Add helper script to generate feature combinations for case study

danjujan committed:
Merge branch 'f-Classification' into jan-dev

danjujan committed:
Use VaRA new confusion matrix implementation

danjujan committed:
Fix mypy + pylint errors

vulder committed:
Implements accessor functions

vulder committed:
Adds stringify function

vulder committed:
Merge branch 'vara-dev' into f-Classification

vulder committed:
Cleans up typing

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Merge branch 'f-Classification' into jan-dev

danjujan committed:
Fix parsing of xz by ignoring regions outside of function of interest

danjujan committed:
Fix parsing of xz. Do not replace root region with other. Merge instead.

danjujan committed:
Warn when vara instruction is ignored, because corresponding file was not found

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Fix mypy  errors

danjujan committed:
Fix failing workload commands because I added a workload in xz

danjujan committed:
Add script to dump vara feature to command line option mapping

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Handle absolute paths in VaRA instructions

danjujan committed:
Use feature command line option mappings specified in feature model

danjujan committed:
Fix

danjujan committed:
RepositoryAtCommit force checkout to workaround conflicts bug

danjujan committed:
Introduce FILE_ROOT Code Region to be able to parse code regions outside functions

danjujan committed:
Fix mypy

danjujan committed:
Skip test if llvm-profdata is not found

danjujan committed:
Reduce FPs in confusion matrices by ignoring gap regions when they do not have vara_instrs

danjujan committed:
Skip tests if vara_feature is not installed

danjujan committed:
Merge same code region instances together

danjujan committed:
Code regions can start or end can overlap. Fixes xz cov_show

danjujan committed:
End of code region is not inside code region anymore. Fixes that some GAP Regions had vara_instrs. Improves confusion matrices.

danjujan committed:
Mock CoverageReport._extract_feature_option_mapping in unittests

danjujan committed:
Fix mypy

danjujan committed:
Fix

danjujan committed:
Use requires support

danjujan committed:
Enable requires support for commands

danjujan committed:
Merge branch 'command_requires' into jan-dev

danjujan committed:
Add vscode launch config

danjujan committed:
Fix docstring

danjujan committed:
Ignore missing type informations of benchbuilds command class

danjujan committed:
Fix failing tests

danjujan committed:
Merge branch 'command_requires' into jan-dev

danjujan committed:
Fix test

danjujan committed:
feature_option_mapping.py uses numeric_values now

danjujan committed:
Improve sanity check output when parsing coverage json

danjujan committed:
XZ: decompress, list, test works

danjujan committed:
Fix concurrency issues with BC File cache

danjujan committed:
Fix pylint + mypy errors

danjujan committed:
Fix mypy

danjujan committed:
Add case_studies

danjujan committed:
Use executable in Command instead of required

danjujan committed:
Fix mypy errors

danjujan committed:
Fix mypy errors

danjujan committed:
Fix mypy errors

danjujan committed:
Coverage Plot: Plot feature annotations without color also to directory

danjujan committed:
Fix mypy findings

danjujan committed:
Fix mypy findings

danjujan committed:
Plot confusion matrix to latex table file

danjujan committed:
Add hrules to confusion matrix latex table file

danjujan committed:
Parse coverage report files in parallel

danjujan committed:
Fix mypy findings + make confusion_matrix output deterministic

danjujan committed:
CoveragePlot: Make it possible to disable workarounds

danjujan committed:
Make feature_option_mapping python 3.9 compatible

danjujan committed:
Try to get xz running on slurm

danjujan committed:
Fix pip system-wide install failing when using fuse-overlayfs

danjujan committed:
Merge branch 'fuse-overlayfs' into jan-dev

danjujan committed:
Fix unittests

danjujan committed:
Fix unittest

danjujan committed:
Merge branch 'fuse-overlayfs' into jan-dev

danjujan committed:
Fix pylint

danjujan committed:
Fix linking xz in debian 10 container when compiling with profiling instructions

danjujan committed:
Drop unnecessary LLVM_ENABLE_PIC cmake variable again

danjujan committed:
Use vara-feature to get all possible configs in feature_combinations.py

danjujan committed:
feature_combinations.py outputs options instead of feature names

boehmseb committed:
Use virtualenv inside container.

This avoids the long-standing issue of not being able to update pip inside containers.

boehmseb committed:
Remove system python from container image.

System python wasn't used anyways since we ship our own more up-to-date python installation.

boehmseb committed:
Fix test

danjujan committed:
Merge remote-tracking branch 'origin/f-ContainerVirtualenv' into jan-dev

danjujan committed:
Revert container fixes for fuse-overlayfs

danjujan committed:
Extract feature model with vara-feature and use it for presence condtion simplification

danjujan committed:
Fix mypy errors

danjujan committed:
Fix test

danjujan committed:
Fix test

danjujan committed:
Fix test

danjujan committed:
Remove patch from plot tests

danjujan committed:
Fix pylint

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Fix hanging when trying to plot

danjujan committed:
FeatureModelFromula now has prefixed numbers to support xz's presets

danjujan committed:
Cache minimize function

danjujan committed:
Incorporate feedback

danjujan committed:
Merge branch 'command_requires' into jan-dev

danjujan committed:
Merge branch 'vara-dev' into command_requires

danjujan committed:
Cleanup

danjujan committed:
Fix pylint

danjujan committed:
Incorporate feedback

danjujan committed:
Merge branch 'command_requires' into jan-dev

danjujan committed:
Fix

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Merge branch 'jan-dev' of github.com:se-sic/VaRA-Tool-Suite into jan-dev

danjujan committed:
Fix mypy

danjujan committed:
Fix mypy

danjujan committed:
Use FeatureModel for presence condition simplification

danjujan committed:
Merge branch 'command_requires' into jan-dev

danjujan committed:
Fix tests + mypy

danjujan committed:
Fix mypy

danjujan committed:
Fix makedir race condition

danjujan committed:
Cleanup test

danjujan committed:
Add ReducedFeatureModel case study for xz

danjujan committed:
Fix ReducedFeatureModel case study for xz

danjujan committed:
Merge branch 'vara-dev' into jan-dev

Commit:
Update

danjujan committed:
Fix end of line

danjujan committed:
Disable GC temporarily for plotting

danjujan committed:
Precommit

danjujan committed:
Pylint

danjujan committed:
Precommit

danjujan committed:
Debug Timeout error

danjujan committed:
Annotate covered notes later

danjujan committed:
Fix timeout value

danjujan committed:
Avoid RecursionError in expr_to_str by simplifying earlier

danjujan committed:
Use prctl syscall to kill child processes if parent dies

danjujan committed:
Optimized map

danjujan committed:
Switch back to ProcessPool

danjujan committed:
Initialize ProcessPool once, improve chunking

danjujan committed:
Increase recursion limit for expr_to_str

danjujan committed:
Simplify dont care bit logic

danjujan committed:
Fix mypy

danjujan committed:
Ignore disabled arguments for configuration option with argument

danjujan committed:
Fix missing FlushTO_Values in feature_option_mapping

danjujan committed:
Ensure we parse expressions to DNF + use exprvar to remove xz preset workaround

danjujan committed:
Use cudd BDDs instead of pyedas espresso_truthtables for presence condition simplification

danjujan committed:
Add dd as dependecy for cudd BDDs

danjujan committed:
Use threads for _cov_segments_file

danjujan committed:
Add timeout arg to optimized_map

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Move dd to requirements.txt

danjujan committed:
Update dd entry in requirements.txt

danjujan committed:
Try global-option instead of install-option

danjujan committed:
Switch to --config-setting

danjujan committed:
Test in requirements.txt

danjujan committed:
Fix version in requirements.txt

danjujan committed:
myscripts: fix regex for values list

danjujan committed:
Support multiple workloads for same extra args in llvm_coverage_experiment

danjujan committed:
Add picosat support

danjujan committed:
Add minimal picosat case_stuy file

danjujan committed:
Spawn only as much threads as needed + shutdown workers

danjujan committed:
Don't use threads

danjujan committed:
Install dd manually

danjujan committed:
Update requirements.txt

danjujan committed:
Update setup.py

danjujan committed:
Update action.yml

danjujan committed:
Update action.yml

danjujan committed:
Update action.yml

danjujan committed:
Use spawn to potentially fix deadlock by using spawn instead of fork in Process Pool + implement espresso_expr myself

danjujan committed:
Add container support to picosat

danjujan committed:
Fix ci

danjujan committed:
Fix mypy + restore old cov_segments code before thread ready

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Fix mypy

danjujan committed:
Better output messages

danjujan committed:
Fix mypy

danjujan committed:
Another test for pyeda

danjujan committed:
Improve time measurements

danjujan committed:
Add picosat reduced case study

danjujan committed:
picosat, do not allow exist status 0, to detect invalid configurations in feature Model

danjujan committed:
Add UNKNOWN workload to picosat

danjujan committed:
Add coverage_feature_set to plot for easier reasoning

danjujan committed:
Rudimentary support for ect

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Add workloads to ect

danjujan committed:
Make ect a configuration specific project

danjujan committed:
Add fix for compiling ect under debian 10

danjujan committed:
Switch to ect reduced feature model

danjujan committed:
Fix pylint

danjujan committed:
Fix cov_show for ect: ingore buggy llvm json data

danjujan committed:
Switch FeaturePerfCSCollection to MSMR

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Fix moved VCommand

danjujan committed:
MSMR case study remove duplicate configs

danjujan committed:
Fix VCommand import

danjujan committed:
Merge branch 'jan-dev' of github.com:se-sic/VaRA-Tool-Suite into jan-dev

danjujan committed:
Incorporate upstream changes

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Correct workload_commands_tags test

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Fix has_features for no features annotated

danjujan committed:
Improve LaTeX tables: added total + all-coverage, all-vara, all-both

danjujan committed:
Add tests for all-coverage and all_vara

danjujan committed:
Improve table generation

danjujan committed:
Fix pylint + mypy

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Switch back to SFI + small improvements for tables

danjujan committed:
Add midrule before TOTAL in result tables + comment out "all-" rows

danjujan committed:
Rework plotting code + ignore code regions workaround

danjujan committed:
Simplify CoverageReports feature report cache

danjujan committed:
Plot thresholds results combined in one table

danjujan committed:
Add ignore parsing code reggions for MSMR

danjujan committed:
Prepare gzip for experiment

danjujan committed:
Switch to gzip case study to reduced

danjujan committed:
Add bzip2 reduced case study

danjujan committed:
Added specificity to tables + added ignore parsing code regions for bzip2, gzip, picosat, xz

danjujan committed:
Restore workarounds

danjujan committed:
Regenerater case studies after vara-feature fixed

danjujan committed:
Dont fail if we use --stdout args

danjujan committed:
Update xz configs after ensuring --compress is always generated by Feature Model

danjujan committed:
Merge branch 'vara-dev' into jan-dev

danjujan committed:
Fix import in experiment

danjujan committed:
Fix mypy + pylint

danjujan committed:
Fix pre-commit

danjujan committed:
Fix broken merge

danjujan committed:
Add SynthIPRuntime support

danjujan committed:
Add workload.small again to fix unittest

danjujan committed:
Fix tests  +  ignore feature-dependent functions in SynthIPRuntime

danjujan committed:
Add SynthDARecursion support

danjujan committed:
Add SynthDADynamicDispatch, SynthOVInsideLoop

danjujan committed:
Add SynthSA support

danjujan committed:
Fix pylint

danjujan committed:
Plot instrs for FNs and FPs in confusion matrices

danjujan committed:
Fix mypy

danjujan committed:
Get rid of duplicates in feature_combinations generation + Update reduced xz casestudy

danjujan committed:
Get rid of duplicates in feature_model_formula

danjujan committed:
Use ConfigParams in more projects

danjujan committed:
Reduce duplicates in gzip case study

danjujan committed:
Get SynthDADynamicDispatch working

danjujan committed:
Use FeatureModel when generating matrices...

danjujan committed:
Add alternative options workaround for SynthIPRuntime and SynthSAContextSensitivity

danjujan committed:
Add excluded options workaround for bzip2

danjujan committed:
Add excluded options workaround for xz
